community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/fusion/FusionIndexReader.java

```diff
@@ -100,6 +107,42 @@ public PrimitiveLongIterator query( IndexQuery... predicates ) throws IndexNotAp
     }
 
     @Override
+    public void query( IndexProgressor.NodeValueClient cursor, IndexOrder indexOrder, IndexQuery... predicates )
+            throws IndexNotApplicableKernelException
+    {
+        if ( predicates.length > 1 )
+        {
+            luceneReader.query( cursor, indexOrder, predicates );
+            return;
+        }
+
+        if ( predicates[0] instanceof ExactPredicate )
+        {
+            ExactPredicate exactPredicate = (ExactPredicate) predicates[0];
+            selector.select( nativeReader, luceneReader, exactPredicate.value() ).query( cursor, indexOrder, predicates );
+            return;
+        }
+
+        if ( predicates[0] instanceof NumberRangePredicate )
+        {
+            nativeReader.query( cursor, indexOrder, predicates[0] );
+            return;
+        }
+
+        // todo: There will be no ordering of the node ids here. Is this a problem?
```

Comment:
If we provide an `ExistsPredicate` with a specified `IndexOrder` we need to fail if we cannot serve it?